### PR TITLE
Centrifuge App: Tranche metadata update

### PR DESCRIPTION
**Pull request by onnovisser**
### Description

Saves tranche metadata by trancheId and updates the UI to expect it in that way (breaks tranche metadata for all pools until they are resaved). Also persists proxy address to localstorage.

### Approvals

- [ ] Dev

### Impact

Centrifuge App

### CODE REVIEW

File: nft-studio/src/pages/IssuerCreatePool/pinPoolMetadata.ts [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
+  }
+  return `0x${out.join('')}`
+}
+
+function computeTrancheId(trancheIndex: number, poolId: string) {
+  const a = new BN(trancheIndex).toArray('le', 8)
+  const b = new BN(poolId).toArray('le', 8)
+  const data = Uint8Array.from(a.concat(b))
+
+  return toHex(hash(data, 16))
```
Comment (sophialittlejohn) [inline nft-studio/src/pages/IssuerCreatePool/pinPoolMetadata.ts:26]: nice job figuring this out 🎉

File: nft-studio/src/components/Web3Provider.tsx [hunk 2]
```diff
@@ -115,6 +120,11 @@ export const Web3Provider: React.FC = ({ children }) => {
     setProxyAddress(null)
   }, [])
 
+  const selectProxy = React.useCallback((address: string | null) => {
+    setProxyAddress(address)
+    localStorage.setItem('web3PersistedProxy', address ?? '')
```
Comment (sophialittlejohn) [inline nft-studio/src/components/Web3Provider.tsx:125]: Just this seems a bit weird to me. I see we're doing it in other places as well. But why are we potentially setting `''` into localStorage? I would think that we shouldn't set the item if `address === null`

**Review by sophialittlejohn — APPROVED** [review]
nice job 💪  Let's make sure we update the all of metadata quickly once this is merged!

File: nft-studio/src/components/Web3Provider.tsx (hunk 2 above)
Comment (onnovisser) [reply nft-studio/src/components/Web3Provider.tsx:125]: if we're calling `selectProxy` with `null`, we'd need to clear any persisted proxy. the alternative would be
```
if (address) {
    localStorage.setItem('web3PersistedProxy', address)
} else {
    localStorage.removeItem('web3PersistedProxy')
}
```
the current is just a bit more concise and the component treats the empty strings it finds in localStorage as undefined. If you think the more verbose alternative is better/more readable, I'm happy too change it.

Comment (sophialittlejohn) [reply nft-studio/src/components/Web3Provider.tsx:125]: fair enough, for me it's fine as it 😄

**Pull request merged**